### PR TITLE
Add course runs

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

This is a POC for adding course runs.

The Ux can be greatly improved by doing a bit of customization and CSS on the admin template.
An alternative could be to declare a plugin...

## Proposal

- [x] Add a CourseRun model with a foreign key to the course
- [x] Add the course run as an inline on the course admin view
- [x] Remove the active_session key on the course

### CODE REVIEW

**Comment by 2018D** [comment]
I made a comment on issue #329

**Comment by 2018D** [comment]
the active_session key is not required but we do still need a course_key. We could rename the value or leave as is for now.

**Comment by 2018D** [comment]
did a rebase to prep the terrain

**Comment by 2018D** [comment]
How can we retrieve the course_run info after filling the inline form to show it on the html? I haven't been able to figure it out.

**Comment by 2018D** [comment]
work to be done:
[ ] fix publishing course does not copy or show draft course-runs on published course-detail
[ ] factories
[ ] unit testing: course admin and course-detail page to include course-run

Should we consider using the licence structure for the course-run?

**Comment by sampaccoud** [comment]
@2018D I have fix the issues you raised. 
I propose to skip testing the course run custom admin widget/field as it is quite painful with little value and I don't have time now. If you think it's important, please do it or add an issue for someone to do it later?

File: src/richie/apps/courses/admin.py [hunk 1]
```diff
@@ -2,17 +2,40 @@
...
 from cms.extensions import PageExtensionAdmin
 
+from .fields import CourseRunSplitDateTimeField
 from .forms import LicenceFormAdmin
-from .models import Course, Licence, Organization
+from .models import Course, CourseRun, Licence, Organization
+from .widgets import CourseRunSplitDateTimeWidget
+
+
+class CourseRunInline(admin.StackedInline):
```
Comment (sveetch) [inline src/richie/apps/courses/admin.py]: You may try `admin.TabularInline` it is commonly more clean and accessible

Comment (sampaccoud) [reply src/richie/apps/courses/admin.py]: Yes that's what I used at first but it requires some adaptation to fit in a line so as a first approach I switched to stacked inlines...

image

Comment (sveetch) [reply src/richie/apps/courses/admin.py]: Ah yeah, table makes troubleshooting with non text element like the date widtget.

**Comment by 2018D** [comment]
C'est bon pour moi 👍

**Comment by sampaccoud** [comment]
Fixes https://github.com/openfun/richie/issues/329

**Pull request merged**